src/fn/watch-project.js

```diff
@@ -0,0 +1,234 @@
+    let form = uploadForms.formFileMatcher(fileName);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await uploadAppForms([form]);
+            return fileName;
+        });
+        return true;
+    }
+
+    form = convertForms.formFileMatcher(fileName);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await convertAppForms([form]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processAppFormMedia = (formMediaDir, fileName) => {
+    const form = uploadForms.formMediaMatcher(formMediaDir);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await uploadAppForms([form]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processContactForm = (fileName) => {
+    let form = convertForms.formFileMatcher(fileName);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await convertContactForm([form]);
+            return fileName;
+        });
+        return true;
+    }
+
+    form = uploadForms.formFileMatcher(fileName);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await uploadContactForms([form]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processCollectForm = (fileName) => {
+    let form = uploadForms.formFileMatcher(fileName);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await uploadCollectForms([form]);
+            return fileName;
+        });
+        return true;
+    }
+
+    form = convertForms.formFileMatcher(fileName);
+    if (form) {
+        eventQueue.enqueue(async () => {
+            await convertCollectForms([form]);
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const processConfigFiles = (api, fileName) => {
+    if (fileName === APP_SETTINGS_JSON_PATH) {
+        eventQueue.enqueue(async () => {
+            await uploadAppSettings(api);
+            return fileName;
+        });
+        return true;
+    }
+    if (configFileMatcher(fileName)) {
+        eventQueue.enqueue(async () => {
+            await compileAppSettings();
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const watchProject = {
+    watch: async (api, blockFn, callback = {}, uploadStateOnStart = false) => {
+        if (uploadStateOnStart) {
+            await uploadInitialState(api);
+            info('Initial State Uploaded');
+        }
+
+        eventQueue = new Queue({ concurrent: 1, start: true });
+
+        fsEventsSubscription = await watcher.subscribe(environment.pathToProject, async (err, events) => {
+            if (err) {
+                error(err);
+                return;
+            }
+            for (const event of events) {
+                if (event.type !== 'update' && event.type !== 'create') {
```

Comment:
Is watching for deletions planned for the next iteration? For example when a form is deleted.